src/api_client.rs

```diff
@@ -432,21 +433,46 @@ impl Client {
     }
 
     /* node namespace */
-    pub async fn get_node_identity() -> Result<NetworkIdentity, Error> {
-        unimplemented!("")
+    pub async fn get_node_identity(&self) -> Result<NetworkIdentity, Error> {
+        let result: Value<NetworkIdentity> = self.get("/eth/v1/node/identity").await?;
+        Ok(result.data)
     }
 
-    pub async fn get_node_peers(filters: &[PeerDescriptor]) -> Result<Vec<PeerDescription>, Error> {
-        unimplemented!("")
+    pub async fn get_node_peers(
+        &self,
+        filters: &[PeerDescriptor],
+    ) -> Result<PeerDescriptionWithMetadata, Error> {
+        let path = "eth/v1/node/peers";
+        let target = self.endpoint.join(path)?;
+        let mut request = self.http.get(target);
+        let mut states = vec![];
+        let mut directions = vec![];
+        if !filters.is_empty() {
+            for descriptor in filters.iter() {
+                states.push(descriptor.state.to_string());
+                directions.push(descriptor.direction.to_string());
+            }
+        }
+        if !states.is_empty() {
+            request = request.query(&[("state", states.iter().join(","))]);
+        };
+        if !directions.is_empty() {
+            request = request.query(&[("direction", directions.iter().join(","))]);
+        }
```

Comment:
after looking more deeply at the API definition, I think we should instead take two different lists to mirror the specs.

the signature would look like this:

```rust
pub async fn get_node_peers(&self, peer_states: &[PeerState], directions: &[ConnectionOrientation])
```

then, rather than make an additional allocation we can just iterate each one directly:

```rust
if !peer_states.is_empty() {
  request = request.query(&[("state", peer_states.iter().join(","))]);
}
```
and similarly for `directions`

Reply:
done